aiocache/lock.py

```diff
@@ -110,8 +116,15 @@ class OptimisticLock(Generic[CacheKeyType]):
     the one we retrieved when the lock started.
 
     Example usage::
+        from aiocache import ValkeyCache
+
+        from glide import GlideClient, GlideClientConfiguration, NodeAddress
+
+        addresses = [NodeAddress("localhost", 6379)]
+        conf = GlideClientConfiguration(addresses=addresses, database_id=0)
+        client = await GlideClient.create(conf)
```

Comment:
This is going to be an issue for our decorators. Is it possible to create a client object without an async call (i.e. without creating the connections), and then starting that client using `async with`?

Reply:
Sure, we can probably do something like that for now.

I wouldn't describe what I'm asking for as lazy connect though, just moving the connection creation to `__aenter__()`. That way there is more control over when the connection is started.

Although, it probably makes sense to bring the client creation back into the Cache class. We moved the Redis client out of the Cache class, because it took so many different arguments, it was too awkward to maintain correctly. Here, we'll only ever need to pass a config object...

Reply:
Yeah, let's just create the client inside the `GlideCache.__aenter__`. That's going to be a lot easier.